config/initializers/dev_server.rb

```diff
@@ -0,0 +1,4 @@
+require_dependency "#{Rails.root}/lib/dev_server_proxy"
+if Features.hot_loading? && Rails.env.development?
+  Rails.application.config.middleware.use WebpackDevServerProxy, dev_server_host: "localhost:8080"
```

Comment:
so... what if, instead of switching on Rails.env.development, this middleware just varied the host based on environment? Like, it uses localhost:3000 in test and localhist:8080 in dev?